app/src/main/kotlin/com/github/gotify/service/WebSocketService.kt

```diff
@@ -106,10 +104,47 @@ internal class WebSocketService : Service() {
             .onOpen { onOpen() }
             .onClose { onClose() }
             .onBadRequest { message -> onBadRequest(message) }
-            .onNetworkFailure { minutes -> onNetworkFailure(minutes) }
+            .onNetworkFailure { seconds -> onNetworkFailure(seconds) }
             .onMessage { message -> onMessage(message) }
             .onReconnected { notifyMissedNotifications() }
             .start()
+        return connection
+    }
+
+
+    private fun startPushService() {
+        UncaughtExceptionHandler.registerCurrentThread()
+        showForegroundNotification(getString(R.string.websocket_init))
+
+        //observe the network changes
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            (getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager)
+                .registerDefaultNetworkCallback(
```

Comment:
Please unregister the callback when the websocket service is stopped https://developer.android.com/reference/android/net/ConnectivityManager#unregisterNetworkCallback(android.net.ConnectivityManager.NetworkCallback)

Reply:
sorry, I forgot it. I have added the code block to unregist the observer.